File: android-lib/src/androidTest/java/com/kinvey/androidTest/callback/DefaultKinveyReadCallback.java
package com.kinvey.androidTest.callback;

import com.kinvey.android.callback.KinveyReadCallback;
import com.kinvey.androidTest.model.Person;
import com.kinvey.java.model.KinveyReadResponse;

import java.util.List;
import java.util.concurrent.CountDownLatch;

/**
 * Created by yuliya on 09/14/17.
 */

public class DefaultKinveyReadCallback implements KinveyReadCallback<Person> {

    private CountDownLatch latch;
    private KinveyReadResponse<Person> result;
    private Throwable error;

    public DefaultKinveyReadCallback(CountDownLatch latch) {
        this.latch = latch;
    }

    @Override
    public void onSuccess(KinveyReadResponse<Person> result) {
        this.result = result;
        finish();
    }

    @Override
    public void onFailure(Throwable error) {
        this.error = error;
        finish();
    }

    void finish() {
        latch.countDown();
    }

    public KinveyReadResponse<Person> getResult() {
        return result;
    }

    public Throwable getError() {
        return error;
    }
}

File: android-lib/src/androidTest/java/com/kinvey/androidTest/callback/DefaultKinveyReadCallback.kt
package com.kinvey.androidTest.callback

import com.kinvey.android.callback.KinveyReadCallback
import com.kinvey.androidTest.model.Person
import com.kinvey.java.model.KinveyReadResponse
import java.util.concurrent.CountDownLatch

/**
 * Created by yuliya on 09/14/17.
 */

class DefaultKinveyReadCallback(private val latch: CountDownLatch) : KinveyReadCallback<Person> {
    var result: KinveyReadResponse<Person>? = null
        private set
    var error: Throwable? = null
        private set

    override fun onSuccess(result: KinveyReadResponse<Person>?) {
        this.result = result
        finish()
    }

    override fun onFailure(error: Throwable) {
        this.error = error
        finish()
    }

    internal fun finish() {
        latch.countDown()
    }

}